### PR TITLE
test(image_projection_based_fusion): add unit test code for geometry

### PR DESCRIPTION
**Pull request by MasatoSaeki**
## Description
Add unit test for below in image_projection_basef_fusion package
- `objectToVertices`
- `boundingBoxToVertices`
- `cylinderToVertices`
- `transformPoints`
- `is_inside`
- `sanitizeROI`

## Tests performed
```
colcon test --merge-install --packages-select image_projection_based_fusion --event-handlers console_cohesion+
```

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/image_projection_based_fusion/test/test_geometry.cpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  // Test at Shape::BOUNDING_BOX
+  geometry_msgs::msg::Pose pose;
+  pose.position.x = 1.0;
+  pose.position.y = 2.0;
+  pose.position.z = 3.0;
+  const double q_scalar = std::sin(M_PI / 12);
+  pose.orientation.x = 0.0;
+  pose.orientation.y = 0.0;
+  pose.orientation.z = q_scalar;
+  pose.orientation.w = std::cos(M_PI / 12);
```
Comment (kminoda) [inline perception/image_projection_based_fusion/test/test_geometry.cpp:32]: How about this? Using a magic number `M_PI / 12` in two different lines may reduce the readability.
```suggestion
  const double angle = M_PI / 12;
  pose.orientation.x = 0.0;
  pose.orientation.y = 0.0;
  pose.orientation.z = std::sin(angle);
  pose.orientation.w = std::cos(angle);
```

File: perception/image_projection_based_fusion/test/test_geometry.cpp [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+  // Test right-bottom outside pattern
+  inner.x_offset = 31;
+  inner.y_offset = 41;
+  inner.height = 401;
+  inner.width = 301;
+
+  inside_flag = image_projection_based_fusion::is_inside(outer, inner, outer_offset_scale);
+
+  EXPECT_FALSE(inside_flag);
```
Comment (kminoda) [inline perception/image_projection_based_fusion/test/test_geometry.cpp:129]: I would recommend to limit the scope for each test in this case:

```suggestion
  {
    sensor_msgs::msg::RegionOfInterest inner;
    inner.x_offset = 31;
    inner.y_offset = 41;
    inner.height = 399;
    inner.width = 299;
    const double outer_offset_scale = 1.0;
  
    const bool inside_flag = image_projection_based_fusion::is_inside(outer, inner, outer_offset_scale);
  
    EXPECT_TRUE(inside_flag);
  }

  {
    // Test left-top outside pattern
    sensor_msgs::msg::RegionOfInterest inner;
    inner.x_offset = 29;
    inner.y_offset = 39;
  
    const bool inside_flag = image_projection_based_fusion::is_inside(outer, inner, outer_offset_scale);
  
    EXPECT_FALSE(inside_flag);
  }

  {
    // Test right-bottom outside pattern
    sensor_msgs::msg::RegionOfInterest inner;
    inner.x_offset = 31;
    inner.y_offset = 41;
    inner.height = 401;
    inner.width = 301;
  
    const bool inside_flag = image_projection_based_fusion::is_inside(outer, inner, outer_offset_scale);
  
    EXPECT_FALSE(inside_flag);
  }
```

Comment (kminoda) [reply perception/image_projection_based_fusion/test/test_geometry.cpp:129]: (Also true in `test_objectToVertices` and `test_sanitizeROI` as well)

**Review by kminoda — COMMENTED** [review]
Thank you for the PR :tada: 
I left some minor comments on the coding style

File: perception/image_projection_based_fusion/test/test_geometry.cpp [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+TEST(sanitizeROI, test_sanitizeROI)
+{
+  // Test default pattern
+  sensor_msgs::msg::RegionOfInterest roi;
+  roi.x_offset = 10;
+  roi.y_offset = 20;
+  roi.height = 200;
+  roi.width = 100;
+  int height_ = 400;  // image height
+  int width_ = 300;   // image width
```
Comment (yukkysaito) [inline perception/image_projection_based_fusion/test/test_geometry.cpp:141]:
```suggestion
  int height = 400;  // image height
  int width = 300;   // image width
```

File: perception/image_projection_based_fusion/test/test_geometry.cpp (hunk 1 above)
Comment (MasatoSaeki) [reply perception/image_projection_based_fusion/test/test_geometry.cpp]: fixed in [4a4f8f6](https://github.com/autowarefoundation/autoware.universe/pull/7096/commits/4a4f8f68a2516389ee445f40e7d9291e9a52fe1d)

File: perception/image_projection_based_fusion/test/test_geometry.cpp (hunk 2 above)
Comment (MasatoSaeki) [reply perception/image_projection_based_fusion/test/test_geometry.cpp]: fixed in [4a4f8f6](https://github.com/autowarefoundation/autoware.universe/pull/7096/commits/4a4f8f68a2516389ee445f40e7d9291e9a52fe1d)

File: perception/image_projection_based_fusion/test/test_geometry.cpp (hunk 3 above)
Comment (MasatoSaeki) [reply perception/image_projection_based_fusion/test/test_geometry.cpp]: fixed in [be2118e](https://github.com/autowarefoundation/autoware.universe/pull/7096/commits/be2118e505d1b4947ae87b0f954d1cf4d5b8e6c0)

**Review by kminoda — APPROVED** [review]
LGTM :tada:

**Comment by kminoda** [comment]
@MasatoSaeki I forced the DCO CI passed. From next time please signoff for each github commit (by `git commit -m "..." -s`) :+1:
https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/commit-guidelines/#sign-off-your-commits-required-automated

**Comment by MasatoSaeki** [comment]
@kminoda 
I am sorry.
I could not merge this PR  becauce build-and-test-differential in CI was not passed. This problem is solved in [6eba819](https://github.com/autowarefoundation/autoware.universe/pull/7096/commits/6eba81917faa322647d54c3a03a8658a7e7863b4). so Could you force DCO CI passed again?

**Pull request merged**